lls/main.c

```diff
@@ -319,13 +385,25 @@ lls_proc(void *arg)
 
 	while (likely(!exiting)) {
 		/* Read in packets on front and back interfaces. */
-		process_pkts(lls_conf, &net_conf->front,
+		uint16_t num_tx = process_pkts(lls_conf, &net_conf->front,
 			lls_conf->rx_queue_front, lls_conf->tx_queue_front);
 		if (net_conf->back_iface_enabled)
-			process_pkts(lls_conf, &net_conf->back,
+			num_tx += process_pkts(lls_conf, &net_conf->back,
 				lls_conf->rx_queue_back,
 				lls_conf->tx_queue_back);
 
+		if (num_tx > 0 && (lacp_enabled(net_conf, &net_conf->front) ||
+				lacp_enabled(net_conf, &net_conf->back))) {
```

Comment:
There's a bug here. If the first `process_pkts()` send out a packet and the second `process_pkts()` didn't send out a packet, the code will reset the timer for both interfaces.

We should try to skip those ARP and ND gratuitous as much as we can without compromising the performance of Gatekeeper. I think the straightforward solution is to have a LACP timer per interface.